integration-tests/test_registration.py

```diff
@@ -0,0 +1,70 @@
+import os
+import subprocess
+
+import pytest
+
+MACHINE_ID_FILE: str = "/etc/insights-client/machine-id"
+
+
+def test_machineid_only_when_registered(external_candlepin, subman, insights_client, test_config):
+    """`machine-id` is only present when insights-client is registered."""
+    assert not insights_client.is_registered
+    assert not os.path.exists(MACHINE_ID_FILE)
+
+    res = insights_client.run(check=False)
+    assert "This host has not been registered. Use --register to register this host." in res.stdout
+    assert not insights_client.is_registered
+    assert not os.path.exists(MACHINE_ID_FILE)
+
+    subman.register(
+        username=test_config.get("candlepin", "username"),
+        password=test_config.get("candlepin", "password"),
+    )
+    insights_client.register()
+    assert insights_client.is_registered
+    assert os.path.exists(MACHINE_ID_FILE)
+
+    insights_client.unregister()
+    assert not insights_client.is_registered
+    assert not os.path.exists(MACHINE_ID_FILE)
+
+
+@pytest.mark.skip("This has always been failing")
+def test_double_register(external_candlepin, subman, insights_client, test_config):
+    """`--register` fails on registered system."""
+    subman.register(
```

Comment:
same applies here, No need to register subman, take a look here- https://github.com/RedHatInsights/insights-client/blob/master/integration-tests/conftest.py#L22.
you can use this fixture in your test like this https://github.com/RedHatInsights/insights-client/blob/master/integration-tests/test_status.py#L5